client/src/client.rs

```diff
@@ -1,32 +1,23 @@
 use anyhow::bail;
 use async_channel::Receiver;
+use givc_common::pb;
+pub use givc_common::query::{Event, QueryResult};
 use tokio_stream::StreamExt;
 use tonic::transport::Channel;
 use tracing::debug;
 
 use givc_common::address::EndpointAddress;
-use givc_common::pb;
-pub use givc_common::query::{Event, QueryResult};
 use givc_common::types::*;
 
 use crate::endpoint::{EndpointConfig, TlsConfig};
 
 type Client = pb::admin_service_client::AdminServiceClient<Channel>;
 
-#[derive(Debug)]
 pub struct WatchResult {
     pub initial: Vec<QueryResult>,
     // Design defence: we use `async-channel` here, as it could be used with both
     // tokio's and glib's eventloop, and recommended by gtk4-rs developers:
     pub channel: Receiver<Event>,
-
-    task: tokio::task::JoinHandle<()>,
-}
-
-impl Drop for WatchResult {
```

Comment:
I am concerning about remove `Drop` here,   I not sure if task spawned for polling grpc would shut down automatically, if `WatchResult` is destroyed.

Reply:
Without the Drop implementation, the spawned task will run until there is an event, at which point the `recv` will fail and task will terminate.

This was mainly for larger refactoring for helping interoperability with gtk-rs, but turned out it actually made things worse.

My main concern with this is that the `WatchResult` cannot be destructured, but I guess we can live with that.

Reply:
Changed it now to use a tokio mpsc channel for signalling; no need to explicitly implement Drop: when the (last, but in this case only) sender is dropped, the `recv()` -future will complete, and the task will terminate.